exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -116,6 +119,7 @@
   private boolean first = true;
   private int targetRecordCount;
   private final String fileName;
+  private Set<Path> currSpillDirs = new TreeSet();
```

Comment:
IntelliJ complains about an "unchecked assignment" here. Can you change this to `new TreeSet<>()` instead ?